### PR TITLE
Tooling to auto deploy Kubeflow in codelab test projects

### PR DESCRIPTION
**Pull request by jlewi**
Tooling to setup multiple instances of Kubeflow in codelab test projects.

* Create a Kubeflow cluster where the K8s jobs to do deployment will run.

* Modify create_unique_kf_instance.py to support using a kfctl binary to
  deploy Kubeflow as opposed to building from source

* bulk_deploy.py contains the code to create a bunch of kubeflow deployments
  1 per project given by a csv file

* Add a utility function to wait for all K8s jobs with a given label
  to finish

* codelabs/codelab-admin contains the KFApp for the kubeflow cluster

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/518)

### CODE REVIEW

File: codelabs/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+1. Ensure the Google Sheet containing the code lab account has a field project
+
+   * With the last batch of projects it looked like the convention was 
+
+     ```
+     USER=devstar${NUMBER}@gcplab.me
+     PROJECT=kf-test-${NUMBER}
+     ```
+
+1. Export the Google Sheet containing the code lab accounts to CSV
```
Comment (zhenghuiwang) [inline codelabs/README.md]: Can you add more detailed on this step, e.g. how the CSV is formatted?

File: codelabs/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+1. Modify [bulk-deploy.yaml](bulk-yaml.yaml) to configure a K8s job to run bulk deployment
+
+   * Set the following command line arguments in the YAML file
+
+     * **projects_path** Change this to the path of the GCS file in the previous step
+
+1. Launch a K8s job running bulk deploy
+
+   ```
+   kubectl create -f bulk-deploy.yaml
```
Comment (zhenghuiwang) [inline codelabs/README.md]: Do we need a bulk-delete?

File: py/kubeflow/testing/bulk_deploy.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        if k.lower() == "username":
+          user = v
+
+      if not project or not user:
+        logging.info("Skipping row %s; could not get project and user", row)
+        continue
+
+      logging.info("Processing project=%s user=%s", project, user)
+
+      # Load a new copy of the job template on each run
```
Comment (zhenghuiwang) [inline py/kubeflow/testing/bulk_deploy.py]: Can we extract line 118-160 to a func `create_job_yaml(...)`?

File: codelabs/README.md (hunk 2 above)
Comment (jlewi) [reply codelabs/README.md]: Probably long term. Filed kubeflow/testing#520

File: py/kubeflow/testing/bulk_deploy.py (hunk 3 above)
Comment (jlewi) [reply py/kubeflow/testing/bulk_deploy.py]: Done.

File: codelabs/README.md (hunk 1 above)
Comment (jlewi) [reply codelabs/README.md]: Done.

**Comment by jlewi** [comment]
@zhenghuiwang we are checking in the full ${KFAPP} which includes the directories ${KFAPP}/kustomize which contains the changes to the kustomize manifests specific to the application. This is the recommended practice since those manifests could potentially contain customizations not in kfctl_gcp_iap.yaml

**Comment by zhenghuiwang** [comment]
> @zhenghuiwang we are checking in the full ${KFAPP} which includes the directories ${KFAPP}/kustomize which contains the changes to the kustomize manifests specific to the application. This is the recommended practice since those manifests could potentially contain customizations not in kfctl_gcp_iap.yaml

Thanks!

/lgtm

**Comment by jlewi** [comment]
Prow says merge is blocked because of invalid OWNERs file; I'm going to try removing it.

**Pull request merged**